### PR TITLE
Add note loading boundary

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Note loading

### Description
This PR adds a note loading boundary that fetches a note from the backend and saves it into the redux state before the editor is shown.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Comment by ErikMichelson** [comment]
For better understanding some screenshots:

**Before this PR**
image

**With this PR**
image

**Normal 404 route (did not change by this PR)**
image

File: src/components/common/note-loading-boundary/hooks/use-load-note-from-server.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  return useAsync(async () => {
+    if (id === undefined) {
+      throw new Error('Invalid id')
+    }
+    try {
+      const noteFromServer = await getNote(id)
+      setNoteDataFromServer(noteFromServer)
+    } catch (error: unknown) {
+      throw new Error('Error while fetching note from server.\n' + (error as Error).message)
```
Comment (ErikMichelson) [inline src/components/common/note-loading-boundary/hooks/use-load-note-from-server.ts:29]: This should not be hardcoded if displayed in the UI. We've got a translation key for that purpose.

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

**Comment by mrdrogdrog** [comment]
@ErikMichelson 

This is.. i don't wanna say "intended" but at least known. This pull requests solves a technical problem. The ui/ux situation is another topic and should be solved when we finally have a design approach.

**Review by DerMolly — APPROVED** [review]

File: src/components/common/note-loading-boundary/note-loading-boundary.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export const NoteLoadingBoundary: React.FC<PropsWithChildren<unknown>> = ({ children }) => {
+  const { error, loading } = useLoadNoteFromServer()
+
+  if (loading) {
+    return <LoadingScreen />
+  } else if (error) {
+    return (
+      <CommonErrorPage
+        titleI18nKey={'noteLoadingBoundary.errorWhileLoadingContent'}
+        descriptionI18nKey={error.message}></CommonErrorPage>
```
Comment (ErikMichelson) [inline src/components/common/note-loading-boundary/note-loading-boundary.tsx:29]: Is there anything prohibiting this?

```suggestion
        descriptionI18nKey={error.message} />
```

**Comment by mrdrogdrog** [comment]
@ErikMichelson huh. interesting. Usually eslint complains when this happens. :shrug:

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**